src/Module/OStatus/PubSubHubBub.php

```diff
@@ -0,0 +1,172 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\OStatus;
+
+use Friendica\App;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Database\Database;
+use Friendica\Model\PushSubscriber;
+use Friendica\Module\Response;
+use Friendica\Network\HTTPClient\Capability\ICanSendHttpRequests;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+/**
+ * An open, simple, web-scale and decentralized pubsub protocol.
+ *
+ * Part of the OStatus stack.
+ *
+ * See https://pubsubhubbub.github.io/PubSubHubbub/pubsubhubbub-core-0.4.html
+ *
+ * @version 0.4
+ */
+class PubSubHubBub extends \Friendica\BaseModule
+{
+	/** @var IManageConfigValues */
+	private $config;
+	/** @var Database */
+	private $database;
+	/** @var ICanSendHttpRequests */
+	private $httpClient;
+
+	public function __construct(ICanSendHttpRequests $httpClient, Database $database, IManageConfigValues $config, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->config     = $config;
+		$this->database   = $database;
+		$this->httpClient = $httpClient;
+	}
+
+	protected function post(array $request = [])
+	{
+		// PuSH subscription must be considered "public" so just block it
+		// if public access isn't enabled.
+		if ($this->config->get('system', 'block_public')) {
+			throw new HTTPException\ForbiddenException();
+		}
+
+		// Subscription request from subscriber
+		// https://pubsubhubbub.github.io/PubSubHubbub/pubsubhubbub-core-0.4.html#rfc.section.5.1
+		// Example from GNU Social:
+		// [hub_mode] => subscribe
+		// [hub_callback] => http://status.local/main/push/callback/1
+		// [hub_verify] => sync
+		// [hub_verify_token] => af11...
+		// [hub_secret] => af11...
+		// [hub_topic] => http://friendica.local/dfrn_poll/sazius
+
+		$hub_mode         = $request['hub_mode']         ?? '';
+		$hub_callback     = $request['hub_callback']     ?? '';
+		$hub_verify_token = $request['hub_verify_token'] ?? '';
+		$hub_secret       = $request['hub_secret']       ?? '';
+		$hub_topic        = $request['hub_topic']        ?? '';
+
+		// check for valid hub_mode
+		if ($hub_mode === 'subscribe') {
+			$subscribe = 1;
+		} elseif ($hub_mode === 'unsubscribe') {
+			$subscribe = 0;
+		} else {
+			$this->logger->notice("Invalid hub_mode=$hub_mode, ignoring.");
+			throw new HTTPException\NotFoundException();
+		}
+
+		$this->logger->info("$hub_mode request from " . $_SERVER['REMOTE_ADDR']);
```

Comment:
Include src\App\Request in this class and than:

```suggestion
		$this->logger->info("hub_mode request details.", ['from' => $this->request->getRemoteAddress(), 'mode' => $hub_mode]);
```